### PR TITLE
docs: add link to the exported_private_dependencies lint

### PR DESCRIPTION
**Pull request by hi-rustin**
It would be a lot quicker for us to have a look around.

### CODE REVIEW

File: src/doc/src/reference/unstable.md [hunk 1]
```diff
@@ -294,7 +294,7 @@ performs the same actions as a regular `build.rs` script would perform.
 
 The 'public-dependency' feature allows marking dependencies as 'public'
 or 'private'. When this feature is enabled, additional information is passed to rustc to allow
-the 'exported_private_dependencies' lint to function properly.
+the [exported_private_dependencies](https://doc.rust-lang.org/rustc/lints/listing/warn-by-default.html#exported-private-dependencies) lint to function properly.
```
[the remark this answers is not in the document]
Comment (hi-rustin) [reply src/doc/src/reference/unstable.md]: Updated. Thanks!

[the remark this answers is not in the document]
Comment (hi-rustin) [reply src/doc/src/reference/unstable.md]: I am not sure how to test it.

**Review by weihanglo — APPROVED** [review]

File: src/doc/src/reference/unstable.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (weihanglo) [reply src/doc/src/reference/unstable.md]: We'll see 😆.

**Comment by weihanglo** [comment]
Thanks!

@bors r+

**Pull request merged**